pkg/providers/common/provider.go

```diff
@@ -1,4 +1,4 @@
-package providers
+package common
```

Comment:
This package renaming sounds somehow confusing to me, and I'd prefer to avoid it. The name `providers` clearly conveyed the scope of the package, whereas `common` is too generic and not related to the provider concept

Reply:
The goal is to encapsulate all aws code files in a package for clarity. I reorganizing the providers package so that there won't be import cycle. The "providers/common" package is meant to have code which other provider packages are importing hereby solving this issue. If you prefer i can place all the files in "providers" package or even merge them to one, but i think it will hurt the clarity and structure

Reply:
I didn't check it but maybe an alternative approach could be to simply have the `aws.go` (provider) file in the `pkg/providers` and then the rest of the additional code in the `aws` subfolder. We've been always kept all the providers at the same level, and this work well also for the clarity (even though there were definitely simpler than this new one)